### PR TITLE
Feature/ts 168/integration tests

### PR DESCRIPTION
**Pull request by trevrosen**
## What
Adds live tests for:

* Alerts
* Charts
* Metrics
* Spaces
* Services

## Why
So that we're regression testing the client against the production service.

### CODE REVIEW

File: _live-tests/main_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+func TestMain(m *testing.M) {
+	token := os.Getenv("APPOPTICS_TOKEN")
+	if token == "" {
+		log.Fatal("set APPOPTICS_TOKEN in the environment")
+	}
+
+	client = appoptics.NewClient(token)
+	// Uncomment the below to see response status/body stdout while tests run
+	//client = appoptics.NewClient(token, appoptics.SetDebugMode())
```
Comment (akahn) [inline _live-tests/main_test.go]: Maybe this should be configurable via ENV, and be enabled for CI? Or is that too verbose?

File: _live-tests/alerts_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	})
+
+	t.Run("DisassociateFromService", func(t *testing.T) {
+		err := client.AlertsService().DisassociateFromService(alert.ID, service.ID)
+		assert.Nil(t, err)
+	})
+
+	t.Run("Update", func(t *testing.T) {
+		alert.Name = "other.name"
+		err := client.AlertsService().Update(alert)
```
[the remark this answers is not in the document]
Comment (trevrosen) [reply _live-tests/alerts_test.go]: Correct -- those are handled in unit tests that work with error payload fixtures.

File: _live-tests/main_test.go (hunk 1 above)
Comment (trevrosen) [reply _live-tests/main_test.go]: Yeah I think that's a little too verbose -- it dumps entire request bodies for each request, which can make it hard to read. I'd vote against making it the default, but I'm not against an available ENV setting for it.

File: _live-tests/alerts_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	})
+
+	t.Run("Retrieve", func(t *testing.T) {
+		fetchedAlert, err := client.AlertsService().Retrieve(alert.ID)
+		assert.Nil(t, err)
+		assert.Equal(t, alert.Name, fetchedAlert.Name)
+	})
+
+	t.Run("AssociateToService", func(t *testing.T) {
+		err := client.AlertsService().AssociateToService(alert.ID, service.ID)
```
[the remark this answers is not in the document]
Comment (trevrosen) [reply _live-tests/alerts_test.go]: oh that shed sailed months ago 😸

**Pull request merged**